analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTests.cs

```diff
@@ -0,0 +1,126 @@
+
+            var sut = new ProjectConfigReader(options);
+
+            sut.AnalysisConfigPath.Should().Be(@"c:\foo\bar\.sonarqube/conf/SonarQubeAnalysisConfig.xml");
+            sut.ProjectPath.Should().Be(@"C:\foo\bar\AwesomeBankWeb.CSharp/AwesomeBankWeb.CSharp.csproj");
+            sut.FilesToAnalyzePath.Should().Be(@"c:\foo\bar\.sonarqube\conf/0/FilesToAnalyze.txt");
+            sut.OutPath.Should().Be(@"C:\foo\bar\.sonarqube/out/0");
+            sut.ProjectType.Should().Be(ProjectType.Product);
+            sut.TargetFramework.Should().Be("netcoreapp3.1");
+        }
+
+        [TestMethod]
+        public void WhenProjectConfigHasMissingFilesToAnalyzePath_ReturnsCorrectValue()
+        {
+            var options = TestHelper.CreateOptions($"ResourceTests\\SonarProjectConfig\\Invalid_MissingFilesToAnalyzePath\\SonarProjectConfig.xml");
+
+            var sut = new ProjectConfigReader(options);
+
+            sut.AnalysisConfigPath.Should().Be(@"c:\foo\bar\.sonarqube\conf\SonarQubeAnalysisConfig.xml");
+            sut.ProjectPath.Should().Be(@"C:\foo\bar\AwesomeBankWeb.CSharp\AwesomeBankWeb.CSharp.csproj");
+            sut.OutPath.Should().Be(@"C:\foo\bar\.sonarqube\out\0");
+            sut.ProjectType.Should().Be(ProjectType.Product);
+            sut.TargetFramework.Should().Be("netcoreapp3.1");
+        }
+
+        [DataTestMethod]
+        [DataRow("Invalid_MissingAnalysisConfigPath")]
+        [DataRow("Invalid_MissingOutPath")]
+        [DataRow("Invalid_MissingProjectPath")]
+        [DataRow("Invalid_MissingProjectType")]
+        [DataRow("Invalid_MissingTargetFramework")]
+
+        public void WhenProjectConfigHasMissingValues_FilesToAnalyzePath_ReturnsCorrectValue(string folder)
+        {
+            var options = TestHelper.CreateOptions($"ResourceTests\\SonarProjectConfig\\{folder}\\SonarProjectConfig.xml");
+
+            var sut = new ProjectConfigReader(options);
+
+            sut.FilesToAnalyzePath.Should().Be(@"c:\foo\bar\.sonarqube\conf\0\FilesToAnalyze.txt");
+        }
+
+        [DataTestMethod]
+        [DataRow("Invalid_DifferentClassName")]
+        [DataRow("Invalid_DifferentNamespace")]
+        [DataRow("Invalid_WrongProjectTypeValue")]
+        [DataRow("Invalid_Xml")]
+        [DataRow(Constants.NuGetLatestVersion)]
+        public void WhenProjectConfigIsValid_FilesToReturnPath_ReturnNull(string folder)
+        {
+            var options = TestHelper.CreateOptions($"ResourceTests\\SonarProjectConfig\\{folder}\\SonarProjectConfig.xml");
+
+            var sut = new ProjectConfigReader(options);
+
+            sut.AnalysisConfigPath.Should().BeNull();
+            sut.ProjectPath.Should().BeNull();
+            sut.FilesToAnalyzePath.Should().BeNull();
+            sut.OutPath.Should().BeNull();
+            sut.ProjectType.Should().BeNull();
+            sut.TargetFramework.Should().BeNull();
+        }
+    }
```

Comment:
Let's add _blind_ test similar to https://github.com/SonarSource/sonar-scanner-msbuild/pull/962 / `WriteProjectConfigFileTests.cs` / `Execute_AllPropertiesAreSet` ensuring that we have a test case for each new property we add in the future.